packages/edge-gateway/src/gateway.js

```diff
@@ -152,22 +152,28 @@ export async function gatewayIpfs(request, env, ctx, options = {}) {
 
     options.onRaceResolution &&
       options.onRaceResolution(winnerGwResponse, gatewayReqs, cid)
+
+    // Add response header
+    const raceResponse = getTransformedResponseWithCustomHeaders(
```

Comment:
I would add the headers after the cache...incase we need to remove this.

Reply:
That means we need to transform the cached response too. We can start with that and then make it like this if we want to proceed